ecommerce/views.py

```diff
@@ -285,3 +385,20 @@ def _update_coupons(basket, product_version, coupons):
                 product_version.product, basket.user, auto_only=True
             )
     return coupon_version
+
+
+def coupon_code_csv_view(request, version_id):
+    """View for returning a csv file of coupon codes"""
+    if not (request.user and request.user.is_staff):
+        raise PermissionDenied
+    coupon_payment_version = get_object_or_404(CouponPaymentVersion, id=version_id)
+    response = HttpResponse(content_type="text/csv")
```

Comment:
Suggestion: https://github.com/mjumbewu/django-rest-framework-csv might simplify a lot of this manual response writing.

Reply:
I looked into that but preferred to avoid another 3rd-party dependency since this view only takes a few lines of code, didn't feel it really fit in as an API url either since it's just a download link for admins.